src/diff/diffDirectories.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import path from "path";
+import _ from "lodash";
+import { diffRaml, RamlDiff } from "./diffProcessor";
+import { ramlToolLogger } from "../common/logger";
+
+/**
+ * Extracts all the RAML file names from the specified config file.
+ *
+ * @param configPath - Target config file
+ *
+ * @returns An array of `assetId/fileName` for every api in the input file
+ */
+export function listRamlsFromConfig(configPath: string): string[] {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const config = require(configPath);
+  const ramls = [];
+  for (const apiFamily of Object.keys(config)) {
+    for (const api of config[apiFamily]) {
+      ramls.push(path.join(api.assetId, api.fatRaml.mainFile));
+    }
+  }
+
+  return ramls;
+}
+
+/**
+ * Compares two arrays and returns their intersection and symmetric difference.
+ *
+ * @param leftArr - One of the arrays being compared
+ * @param rightArr - The other array being compared
+ *
+ * @returns An array each for all the common, exclusive to left and exclusive to
+ * right elements.
+ */
+function compareArrays<T>(
```

Comment:
Hmm. Not sure about the name "compareArrays" but I am at a loss for a better name.